pkgs/development/yottadb/default.nix

```diff
@@ -0,0 +1,225 @@
+{
+  lib,
+  config,
+  makeWrapper,
+  stdenv,
+  writeText,
+  writeScript,
+
+  cmake,
+  coreutils-full,
+  elfutils,
+  fetchFromGitLab,
+  git,
+  glibcLocales,
+  libxcrypt,
+  icu,
+  ncurses,
+  openssl,
+  pkg-config,
+  tcsh,
+  util-linux,
+  zlib
+}:
+
+# To test this package, go to $NIXPKGS/nixos/tests and run:
+#   $ $(nix-build -A driverInteractive yottadb.nix)/bin/nixos-test-driver -I
+#   >>> test_script()
+
+stdenv.mkDerivation rec {
+  pname = "yottadb";
+  version = "1.38";
+
+  src = fetchFromGitLab {
+    owner = "YottaDB";
+    repo = "DB/YDB";
+    rev = "r${version}";
+    leaveDotGit = true;
+    hash = "sha256-k6kAIdPW75cqdT+dXqjlS8TpT2NjQYBTceDzMvWD/oE=";
+  };
+
+  patches = [
+    ./01-gtmsecshr-relocate.patch
+  ];
+
+  NIX_CFLAGS_COMPILE = "-DYDB_EXTERNAL_SECSHR_PARENT_DIR=\"${ydbSecRunDir}\"";
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = buildInputs ++ [
+    cmake
+    git
+    icu.dev
+    makeWrapper
+    ncurses
+    pkg-config
+    tcsh
+    zlib
+  ];
+
+  buildInputs = [
+    coreutils-full
+    elfutils.dev
+    glibcLocales
+    libxcrypt
+    icu
+    openssl
+    util-linux
+  ];
+
+  # The version number is generated using git-describe
+  # and any code patching makes this version number dirty.
+  # Do not modify the git repo in any way!
+  dontFixCmake = true;
+
+  preConfigure = ''
+    # The following cannot be passed to cmakeFlags directly due
+    # to variable substitution
+    cmakeFlags="$cmakeFlags -DYDB_INSTALL_DIR:STRING=$out/dist"
```

Comment:
Why not using cmakeFlags outside this?

Reply:
How do I reference `$out/dist` from `cmakeFlags`?

Reply:
Just write 
```nix
  cmakeFlags = [
    (lib.cmakeOptionType "filepath" "YDB_INSTALL_DIR" "${placeholder "out'}/dist")
  ];
```